### DISCUSSION TITLE
Chrome like Multilogin

### DISCUSSION THREAD

**Discussion by henzycuong1** [seleniumbase/SeleniumBase#2035] [General]
I have a question, anyone know multilogin, gologin, .... I wondered why they can make a chrome that fake all hardware like a new device. I tried using javascript but on [iphey](https://iphey.com/) my canvas not changed. Anyone have any ideas?

**Comment by mdmintz** [comment]
Are you looking for **UC Mode**?

SeleniumBase **UC Mode** is used for bypassing bot-detection services. It's not the default mode, but it can be activated via `pytest` command-line options, eg: `pytest --uc`, or by passing `uc=True` as an arg for the [Syntax Formats](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md) that are called directly with `python`.

### Here are some existing examples that use ``--uc`` mode:
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/verify_undetected.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_uc_mode.py

### Here's a raw driver example:

```python
from seleniumbase import Driver
import time

driver = Driver(uc=True)
driver.get("https://nowsecure.nl/#relax")
time.sleep(6)
driver.quit()
```

--------

While using **UC Mode**, a SeleniumBase script can browse websites appearing as human:

<img width="568" alt="Undetected SeleniumBase" src="https://user-images.githubusercontent.com/6788579/213031519-0220d19f-e210-4858-b6b5-9253b112efe4.png">

**Reply by henzycuong1** [reply]
Yeah, not only undetected, Multilogin can change my vendor, cpu,.... It's like a new device

**Reply by mdmintz** [reply]
You can use `execute_cdp_cmd()` to change Chrome internal settings. Here's an example that uses it to change the Geo Location: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_geolocation.py

**Reply by henzycuong1** [reply]
Thank sir, I'll try it
